clash-lib/src/Clash/Core/Pretty.hs

```diff
@@ -38,42 +43,83 @@ import Numeric                          (fromRat)
 import Clash.Core.DataCon               (DataCon (..))
 import Clash.Core.Literal               (Literal (..))
 import Clash.Core.Name                  (Name (..))
-import Clash.Core.Term                  (Pat (..), Term (..))
+import Clash.Core.Term                  (Pat (..), Term (..), CoreContext (..), primArg)
 import Clash.Core.TyCon                 (TyCon (..), TyConName, isTupleTyConLike)
 import Clash.Core.Type                  (ConstTy (..), Kind, LitTy (..),
                                          Type (..), TypeView (..), tyView)
 import Clash.Core.Var                   (Id, TyVar, Var (..))
 import Clash.Util
+import Clash.Pretty
 
--- | PrettyPrec printing Show-like typeclass
-class PrettyPrec p where
-  pprPrec :: Monad m => PrettyOptions -> Rational -> p -> m (Doc ann)
-
+-- | Options for the pretty-printer, controlling which elements to hide.
 data PrettyOptions = PrettyOptions
-  { displayUniques    :: Bool
-  , displayTypes      :: Bool
-  , displayQualifiers :: Bool
+  { displayUniques    :: Bool -- ^ whether to display unique identifiers
```

Comment:
Could you put the comments below the field name as outlined in our [non-official official Clash style guide](https://github.com/clash-lang/clash-compiler/blob/feature-styleguide/STYLE.md#top-level-definitions)? :)